flow/connectors/mongo/cdc.go

```diff
@@ -224,9 +231,16 @@ func (c *MongoConnector) PullRecords(
 			return fmt.Errorf("failed to decode change stream document: %w", err)
 		}
 
-		if _, ok := changeDoc["operationType"]; !ok {
+		if op, ok := changeDoc["operationType"]; !ok {
 			c.logger.Warn("operationType field not found")
 			continue
+		} else if _, supported := SupportedOperations[op.(string)]; !supported {
```

Comment:
this logic overlaps with the switch below, this could live in that switch's default case

Reply:
It may make sense to have `SupportedOperations` be `map[string]func()` which handles the cases then, removing unreachable default case

Logic is currently confused since below we check `op.(string)` cast, but up here we panic when it isn't a string